include/flatbuffers/flatbuffers.h

```diff
@@ -84,6 +85,9 @@ typedef uint16_t voffset_t;
 
 typedef uintmax_t largest_scalar_t;
 
+// Pointer to relinquished memory.
+typedef std::unique_ptr<uint8_t, std::function<void(uint8_t */* unused */)>> unique_ptr_t;
```

Comment:
Other possible meaningful names are `buffer_t` or `memory_t`, which less emphasis the *uniqueness*, and more what it represents.